### PR TITLE
keyword refactored for hex-based values

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- refactored keyword to detect hex-based values ``salt = b"\x04\xbc....."``

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] rollback custom benchmark after https://github.com/Samsung/CredData/pull/49
- [x] Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -181,7 +182,7 @@ jobs:
 
     - name: Exclude very huge data
       if: steps.cache-data.outputs.cache-hit == 'true'
-      run: rm -rf data/81cd05d0 data/71455799 data/8cda00f3 data/a09d9e50
+      run: rm -rf data/8* data/7* data/a* data/2* data/0* data/f*  data/b* data/d*
```
Comment (csh519) [inline .github/workflows/benchmark.yml:185]: Isn't it remove too many directories..?
Could you let me know how many directories and files are removed, and how much the coverage is decreased?

File: cicd/benchmark.txt [hunk 2]
```diff
@@ -1,25 +1,25 @@
-DATA: 19434458 valid lines. MARKUP: 74510 items
+DATA: 19434458 valid lines. MARKUP: 74638 items
```
Comment (csh519) [inline cicd/benchmark.txt:1]: Following above `rm` command, i think the DATA count should be decreased..
Could you please check this again?

File: credsweeper/common/constants.py [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-
-class Separator:
-    """Separators collection"""
-    common = "=|:=|:"
-    # All unique non-regex characters from `common`
-    common_as_set = "=:"
-    c = "="
-    java = "="
-    json = ":"
-    php = "=>"
```
Comment (csh519) [inline credsweeper/common/constants.py:21]: Removing unused lines are good :+1:

File: credsweeper/filters/value_pem_pattern_check.py [hunk 4]
```diff
@@ -17,3 +19,4 @@ def __init__(self, config: Config):
         """
         super().__init__(config)
         self.pattern_len = config.pem_pattern_len
+        self.pattern = re.compile(fr"(.)\1{{{str(self.pattern_len - 1)},}}")
```
Comment (csh519) [inline credsweeper/filters/value_pem_pattern_check.py:22]: It seems this line is same as `ValuePatternCheck` one...
BTW isn't it useless from `ValuePemPatternCheck`?

File: cicd/benchmark.txt (hunk 2 above)
Comment (babenek) [reply cicd/benchmark.txt:1]: not for just "benchmark" 
``rm`` is executed only for "performance benchmark"

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (babenek) [reply .github/workflows/benchmark.yml:185]: it is only for performance benchmark to reduce common time less than full benchmark with the scores.

Comment (csh519) [reply .github/workflows/benchmark.yml:185]: I see. I understood.

File: .github/workflows/benchmark.yml [hunk 5]
```diff
@@ -62,7 +62,7 @@ jobs:
     - name: Checkout CredData
       uses: actions/checkout@v3
       with:
-        repository: Samsung/CredData
+        repository: babenek/CredData
```
Comment (csh519) [inline .github/workflows/benchmark.yml:65]:
```suggestion
        repository: Samsung/CredData
```

File: tests/filters/test_value_array_dictionary_check.py [hunk 6]
```diff
@@ -11,11 +11,11 @@ class TestValueArrayDictionaryCheck:
...
+            "name": "Pass",
             "severity": "medium",
             "type": "keyword",
-            "values": ["token"],
-            "filter_type": [VariableNotAllowedPatternCheck.__name__],
+            "values": ["pass"],
+            "filter_type": [ValueArrayDictionaryCheck.__name__],
             "use_ml": True,
             "min_line_len": 0,
             "validations": [],
```
Comment (csh519) [inline tests/filters/test_value_array_dictionary_check.py:21]: Why you changed the `Token` to `Pass`?
Isn't `token_rule()` method testing about `Token`?

File: tests/filters/test_value_not_allowed_pattern.py [hunk 7]
```diff
@@ -1,19 +1,37 @@
...
+            "type": "keyword",
+            "values": ["pass"],
+            "filter_type": [ValueNotAllowedPatternCheck.__name__],
+            "use_ml": True,
+            "min_line_len": 0,
+            "validations": [],
+            "doc_available": True,
+        }
+        rule = Rule(config, token_rule_without_filters)
+        return rule
```
Comment (csh519) [inline tests/filters/test_value_not_allowed_pattern.py:25]: As same as here.

Comment (babenek) [reply tests/filters/test_value_not_allowed_pattern.py:25]: https://github.com/babenek/CredSweeper/blob/6e8057c5a272b31875ab80c502996d023ffbd48f/tests/filters/conftest.py#L8

**Review by csh519 — APPROVED** [review]
Okay. Salt values can be detected now.

LGTM :+1:

File: tests/filters/test_value_array_dictionary_check.py (hunk 6 above)
Comment (csh519) [reply tests/filters/test_value_array_dictionary_check.py:21]: I see thank you for explanation.

**Review by xDizzix — APPROVED** [review]
+

**Pull request merged**